app.py

```diff
@@ -39,75 +52,109 @@ def daily():
-	
-	
-	graph_data = OrderedDict() # The data rearranged for usage in the render template (see format below)
-	""" graph_data {
-			"Battery": {
-				"Voltage": {"1579970374000": 423, "1579970379000": 419, ...},
-				"Current": {"1579970374000": 46, "1579970379000": 48, ...},
-				...
-			}, "Solar": ...
+	graph_data = OrderedDict() # The data used in the render template (see format below)
+	""" graph_data = {
+			"Voltage": {"1579970374000": 423, "1579970379000": 419, ...},
+			"Current": {"1579970374000": 46, "1579970379000": 48, ...},
 		} """
 	
+	# Loop through every sensor the current tab should show a reading for
+	for sensor_id in client_format[tab]["lines"]:
 	
-	# Tabs array represents each different graph tab the user can look at
-	tabs = []
-	for tab, tab_data in client_format.items():
-		tabs.append(tab)
-		graph_data[tab] = OrderedDict()
-		print("===== TAB: " + tab + " =====")
-		
-		# Loop through every sensor the current tab should show a reading for
-		for sensor_id in tab_data["lines"]:
-		
-			# Find the info about the sensor
-			sensor = next((item for item in db_format if item["id"] == sensor_id), None)
-			# Ensure the sensor is in the database
-			if sensor is not None and "index" in sensor:
-				graph_data[tab][sensor["name"]] = OrderedDict()
-				print("-- " + str(sensor["index"]) + ": " + sensor_id + " --")
-				
-				# Loop through all the sensor readings for the day being viewed
-				for time, reading in readings.items():					
-					unix = int(date.timestamp() + int(time))*1000 # TODO: Timezone incorrect
-					graph_data[tab][sensor["name"]][unix] = reading[sensor["index"]]
-					print(str(unix) + ": " + str(reading[sensor["index"]]))
+		# Find the info about the sensor
+		sensor = next((item for item in db_format if item["id"] == sensor_id), None)
+		# Ensure the sensor is in the database
+		if sensor is not None and "name" in sensor:
+			graph_data[sensor["name"]] = OrderedDict()
+			#print("-- " + str(sensor["index"]) + ": " + sensor_id + " --")
 			
+			# Loop through all the sensor readings for the day being viewed
+			db_data = db.collection(DATABASE_COLLECTION).document(date_str).collection(sensor["id"]).stream()
+			try:
+				readings = next(db_data).to_dict()["seconds"] # The map within the sensor's document
+			except StopIteration:
+				continue # Skip sensors not in database
+			
+			# Convert keys from strings to ints and sort (conversion required for sort to be correct)
+			# Then convert the sorted list of tuples into two separate lists using zip
+			times, readings = zip(*sorted({int(k) : v for k, v in readings.items()}.items()))
+			
+			# Downsample data if needed
+			if len(readings) > MAX_POINTS:
+				times, readings = min_max_downsample(np.array(times), np.array(readings), MAX_POINTS)
+			
+			for time, reading in zip(times, readings):				
+				unix = int(date.timestamp() + time)*1000
+				graph_data[sensor["name"]][unix] = reading
+				#print(str(unix) + ": " + str(reading))
```

Comment:
Same here - if this needs to be left in the code then it's fine, but otherwise it should probably be removed.